### ISSUE TITLE
Duke2Market results

### ISSUE THREAD

**Issue by dinggd** [LcDog/DomainAdaptiveReID#2]
I have run the code both ways, duke2market and market2duke, with the pre-trained model provided in Drive to find out that the market2duke is a match to the reported numbers while duke2market has a large drop of around 4% in performance (See the results comparison below)
| | Rank-1 | Rank-5 | Rank-10| mAP|
| --- | :---: | :---: | :---: | :---: |
|Reported| 75.8|89.5|93.2|53.7|
|Observed| 71.5|85.0|88.9|48.6|
No change has been made to the training code, can you please throw some light on what the reasons probably are?

**Comment by lsongx** [comment]
@gddingcs would you mind providing the info about the platform & envs you are using? are you using the default parameters provided, e.g. batch size? and maybe sending me a copy of your training log

**Comment by lsongx** [comment]
@gddingcs Our experience is that the results are not stable. The result of duke2market may range from 73 to 78, and the reported result is the mean of 10 repeated results. By the way, training on two GPUs will improve the performance, probably due to asynchronous BN in PyTorch.

**Comment by dinggd** [comment]
> @gddingcs would you mind provide the info about the platform & envs you are using? are you using the default parameters provided, e.g. batch size? and maybe sending me a copy of your training log


Thanks for your prompt reply.

Yup, all parameters are set by default and I'm running on Linux LTS 16.04 with python 2.7.13 and Pytorch 0.4.0. (I came to realize that I have changed some lines by adapting to 0.4.0, however, those only involve casting tensors to list and scalar.)

I'm training with only one GPU and now running a second time to see whether higher performance will pop up.

I did not save the logs but will try log them in future runs.

**Comment by dinggd** [comment]
It would be even better if you can kindly provide the self-training baseline training code! :)

**Comment by lsongx** [comment]
@gddingcs Removing the reranking step and you can get the self-training baseline code. You can simply add a line `return original_dist` after line 42 in `rerank.py`. :)

**Comment by dinggd** [comment]
@LcDog Unfortunately, I ran 3 more times (4 in total) and yielded following results:
Mean AP: 48.6%
CMC Scores    allshots      cuhk03  market1501
  top-1          28.5%       54.0%       71.5%
  top-5          42.9%       77.6%       85.0%
  top-10         50.2%       84.7%       88.9%
Mean AP: 50.9%
CMC Scores    allshots      cuhk03  market1501
  top-1          30.1%       55.9%       73.0%
  top-5          44.9%       79.7%       85.9%
  top-10         52.5%       86.7%       89.7%
Mean AP: 50.3%
CMC Scores    allshots      cuhk03  market1501
  top-1          29.5%       55.6%       72.6%
  top-5          44.4%       79.1%       86.0%
  top-10         51.9%       86.5%       89.8%
Mean AP: 49.7%
CMC Scores    allshots      cuhk03  market1501
  top-1          29.3%       55.1%       72.6%
  top-5          43.6%       78.5%       84.8%
  top-10         51.3%       85.8%       89.8%

**Comment by lsongx** [comment]
@gddingcs Hi, I re-ran the code. Here are my results and training logs. Using two GPUs always perfroms better. Could you provide your training logs and env, for better analyzing your results?

```
CUDNN_VERSION=5.1.10
CUDA_PKG_VERSION=8-0-8.0.61-1
CUDA_VERSION=8.0.61
```
## With 1 GPU
| Job ID | 24312 | 24313 | 24314 | 24315| 24316 |24317|
| --- | :---: | :---: | :---: | :---: | :---: | :---: |
|Rank 1| 72.7|72.8|73.1|72.9|73.8|73.0|

[OneDrive Download Link](https://1drv.ms/u/s!AqzBcxT1FhwGjmvTf1AADW7JRVf2)

## With 2 GPU
| Job ID | 24306 | 24307 | 24308 | 24309| 24310 |24311|
| --- | :---: | :---: | :---: | :---: | :---: | :---: |
|Rank 1| 75.1|75.7|77.5|76.1|77.5|77.3|

[OneDrive Download Link](https://1drv.ms/u/s!AqzBcxT1FhwGjmrOVqAc80h3STAN)

**Comment by dinggd** [comment]
@LcDog Alright, thank you and I'm now running on two gpus!!!

**Comment by dinggd** [comment]
@LcDog I'm closing this issue as the two gpu version gave me 76.0% rank-1 accuracy, thanks!!!

**Comment by lsongx** [comment]
@gddingcs Glad to hear that :)
